syntax = "proto3";
option optimize_for = LITE_RUNTIME;
package Pb.Client;

enum MsgId
{
    id0 = 0;
    ID_PlayerLoginReq       = 1000;
    ID_PlayerLoginRes       = 1001;
}


enum OS_PLATFORM
{
    OS_UNKNOWN     = 0;
    OS_DESKTOP     = 1;
    OS_ANDROID     = 2;
    OS_IOS         = 3;
};

enum LoginAppSdk
{
    UNKNOWN_LAS = 0;
    IDANDPASSWORD = 1;
    WECHAT=2;
    KEY_ENC=3;
    GUEST = 4;
};

message PlayerLoginReq
{
    OS_PLATFORM  osplatform     = 1;
    LoginAppSdk  las            = 2;
    string id                   = 3;
    string pass                 = 4;
    string app_spec             = 5;
    string keys_enc             = 6;
};


message PlayerLoginRes
{
    int32  result           = 1;
    string key              = 2;
    uint64 playerid         = 3;
    string name             = 4;
    int32 sex               = 5;    
}


